api/src/test/java/org/openmrs/validator/VisitValidatorTest.java

```diff
@@ -243,209 +243,241 @@ public void validate_shouldFailIfAnAttributeIsBad() {
-		
+
 		Calendar calendar = Calendar.getInstance();
 		calendar.set(2014, Calendar.FEBRUARY, 2);
-		
+
 		Visit visit = makeVisit(42);
 		visit.setStartDatetime(calendar.getTime());
-		
+
 		calendar.set(2014, Calendar.FEBRUARY, 12);
 		visit.setStopDatetime(calendar.getTime());
-		
+
 		assertTrue(patientHasVoidedVisit(visit.getPatient(), DATE_TIME_2014_02_05_00_00_00_0,
-		    DATE_TIME_2014_02_11_00_00_00_0));
-		
+				DATE_TIME_2014_02_11_00_00_00_0));
+
 		Errors errors = new BindException(visit, "visit");
 		new VisitValidator().validate(visit, errors);
-		
+
 		assertFalse(errors.hasFieldErrors("stopDatetime"));
 	}
-	
+
 	private boolean patientHasVoidedVisit(Patient patient, long startInMillis, long stopInMillis) {
-		
+
 		// To get voided visit from the past, both inactive AND voided visits are queried.
 		//
 		List<Visit> visitList = Context.getVisitService().getVisitsByPatient(patient, true, true);
 		for (Visit visit : visitList) {
 			if (visit.getStartDatetime() != null && visit.getStartDatetime().getTime() == startInMillis
-			        && visit.getStopDatetime() != null && visit.getStopDatetime().getTime() == stopInMillis
-			        && visit.getVoided()) {
+					&& visit.getStopDatetime() != null && visit.getStopDatetime().getTime() == stopInMillis
+					&& visit.getVoided()) {
 				return true;
 			}
 		}
 		return false;
 	}
-	
+
 	/**
-	 * @see VisitValidator#validate(Object,Errors)
+	 * @see VisitValidator#validate(Object, Errors)
 	 */
 	@Test
 	public void validate_shouldPassValidationIfFieldLengthsAreCorrect() {
 		Visit visit = makeVisit(42);
 		visit.setVoidReason("voidReason");
-		
+
 		Errors errors = new BindException(visit, "visit");
 		new VisitValidator().validate(visit, errors);
 		assertEquals(false, errors.hasFieldErrors("voidReason"));
 	}
-	
+
 	/**
-	 * @see VisitValidator#validate(Object,Errors)
+	 * @see VisitValidator#validate(Object, Errors)
 	 */
 	@Test
 	public void validate_shouldFailValidationIfFieldLengthsAreNotCorrect() {
 		Visit visit = makeVisit(42);
 		visit
-		        .setVoidReason("too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text");
-		
+				.setVoidReason(
+						"too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text");
+
 		Errors errors = new BindException(visit, "visit");
 		new VisitValidator().validate(visit, errors);
 		assertEquals(true, errors.hasFieldErrors("voidReason"));
 	}
+
+	@Test
+	public void validate_shouldFailIfStartDateTimeIsBeforeBirthDate() {
```

Comment:
You need to separate these tests i.e the one for estimated birthdate and the one for a non estimated date